File: examples/basic-examples/simple-stack-example-basic-java-fragment/src/main/java/com/zhuinden/navigationexamplefrag/core/navigation/BaseKey.java
package com.zhuinden.navigationexamplefrag.core.navigation;

import android.os.Bundle;
import android.os.Parcelable;

/**
 * Created by Owner on 2017. 06. 29..
 */

public abstract class BaseKey implements Parcelable {
    public String getFragmentTag() {
        return toString();
    }

    public final BaseFragment newFragment() {
        BaseFragment fragment = createFragment();
        Bundle bundle = fragment.getArguments();
        if (bundle == null) {
            bundle = new Bundle();
        }
        bundle.putParcelable("KEY", this);
        fragment.setArguments(bundle);
        return fragment;
    }

    protected abstract BaseFragment createFragment();
}

File: simple-stack-example-basic-kotlin/src/main/java/com/zhuinden/simplestackexamplekotlin/BaseKey.kt
package com.zhuinden.simplestackexamplekotlin

import android.os.Bundle
import android.os.Parcelable

/**
 * Created by Owner on 2017.11.13.
 */
abstract class BaseKey : Parcelable {
    val fragmentTag: String
        get() = toString()

    fun newFragment(): BaseFragment = createFragment().apply {
        arguments = (arguments ?: Bundle()).also { bundle ->
            bundle.putParcelable("KEY", this@BaseKey)
        }
    }

    protected abstract fun createFragment(): BaseFragment
}
